ai-risk-score/src/ai_risk_score.rs

```diff
@@ -0,0 +1,78 @@
+use bs58;
+use ed25519_dalek::ed25519::signature::Signature;
+use ed25519_dalek::PublicKey;
+use ed25519_dalek::Verifier;
+use hex::FromHex;
+use lazy_static::lazy_static;
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+use std::str::FromStr;
+use domichain_sdk::pubkey::Pubkey;
+
+lazy_static! {
+    pub static ref RISK_SCORE_MAP: Arc<RwLock<HashMap<Pubkey, HashMap<Pubkey, RewardData>>>> =
+        Arc::new(RwLock::new(HashMap::new()));
+}
+
+#[derive(Debug, Clone)]
+#[allow(dead_code)]
+pub struct RewardData {
+    pub risk_score: f64
+}
+
+#[allow(dead_code)]
+fn from_base58_str(s: &str) -> Vec<u8> {
+    bs58::decode(s)
+        .into_vec()
+        .expect("Failed to decode base58 string")
+}
+
+#[allow(dead_code)]
+fn verify_signature(data_hex: &str, signature_hex: &str, public_key_hex: &str) -> bool {
+    let data_bytes = Vec::from_hex(data_hex).expect("Failed to decode data hex string");
+    let signature_bytes =
+        Vec::from_hex(signature_hex).expect("Failed to decode signature hex string");
+    let public_key_bytes = from_base58_str(public_key_hex);
+
+    let public_key =
+        PublicKey::from_bytes(&public_key_bytes).expect("Failed to create PublicKey from bytes");
+    let signature =
+        Signature::from_bytes(&signature_bytes).expect("Failed to create Signature from bytes");
+
+    public_key.verify(&data_bytes, &signature).is_ok()
+}
+
+pub fn update_risk_scores(wallet: String, reward_account: String, mut risk_score: f64) {
```

Comment:
This should also be `Pubkey`s:
```suggestion
pub fn update_risk_scores(wallet: Pubkey, reward_account: Pubkey, mut risk_score: f64) {
```

Reply:
https://github.com/Domino-Blockchain/domichain/commit/352d0040488aa62df8df2afefd55e1445b5bb8d3

Reply:
Not fixed